lib/runtime/src/component.rs

```diff
@@ -281,60 +284,47 @@ impl Component {
             self.service_name()
         ); // it happens that in component, hierarchy and service name are the same
 
-        // Register a metrics callback that scrapes component statistics
-        let metrics_callback = Arc::new(move || {
-            // Timeout for scraping metrics from components (in milliseconds)
-            // This value is also used by KV Router metrics aggregator (300ms) and other components
-            const METRICS_SCRAPE_TIMEOUT_MS: u64 = 300;
-
-            // Get the current Tokio runtime handle
-            let handle = tokio::runtime::Handle::try_current()
-                .map_err(|err| anyhow::anyhow!("No Tokio runtime handle available: {}", err))?;
+        // Start a background task that scrapes stats every 5 seconds
+        let m = component_metrics.clone();
+        let c = component_clone.clone();
 
-            let m = component_metrics.clone();
-            let c = component_clone.clone();
-
-            // Create a channel to synchronize with the spawned task
-            let (tx, rx) = std::sync::mpsc::channel::<anyhow::Result<()>>();
+        // Use std::thread for the background task to avoid runtime context issues
+        std::thread::spawn(move || {
+            // Create a new runtime for this background thread
```

Comment:
wanted to double check here - my understanding is that we have primary and secondary runtimes already created - and not sure if we would want to use those or not here - or if we need a new runtime.

@ryanolson , @grahamking - for suggestion here -